src/Pim/Bundle/CatalogBundle/Elasticsearch/Sorter/BaseSorter.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Sorter;
+
+use Pim\Bundle\CatalogBundle\Elasticsearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Exception\InvalidDirectionException;
+use Pim\Component\Catalog\Query\Sorter\Directions;
+use Pim\Component\Catalog\Query\Sorter\FieldSorterInterface;
+
+/**
+ * Base sorter for an Elasticsearch query
+ *
+ * @author    Anaël Chardan <anael.chardan@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class BaseSorter implements FieldSorterInterface
+{
+    /** @var SearchQueryBuilder */
+    protected $searchQueryBuilder;
+
+    /** @var array */
+    protected $supportedFields;
+
+    /**
+     * @param array $supportedFields
+     */
+    public function __construct(array $supportedFields = [])
+    {
+        $this->supportedFields = $supportedFields;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setQueryBuilder($searchQueryBuilder)
+    {
+        if (!$searchQueryBuilder instanceof SearchQueryBuilder) {
+            throw new \InvalidArgumentException(
+                sprintf('Query builder should be an instance of "%s"', SearchQueryBuilder::class)
+            );
+        }
+
+        $this->searchQueryBuilder = $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFieldSorter($field, $direction, $locale = null, $scope = null)
+    {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the sorter.');
+        }
+
+        switch ($direction) {
+            case Directions::ASCENDING:
+                $sortClause = [
+                    $field => [
+                        "order" => $direction,
+                        "missing" => "_first"
```

Comment:
So for this question, this will be harmonized in another PR :) (seen with Delph :))